### ISSUE TITLE
500 error when deleting user (with split databases)

### ISSUE THREAD

**Issue by jnm** [kobotoolbox/kobocat#595]
Possibly due to a post-delete signal?
https://sentry.kbtdev.org/kobo/kobocat/issues/302116
```
DoesNotExist: User matching query does not exist.
  File "django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "django/contrib/admin/options.py", line 618, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
  File "django/utils/decorators.py", line 110, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "django/views/decorators/cache.py", line 57, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "django/contrib/admin/sites.py", line 233, in inner
    return view(request, *args, **kwargs)
  File "django/utils/decorators.py", line 34, in _wrapper
    return bound_func(*args, **kwargs)
  File "django/utils/decorators.py", line 110, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "django/utils/decorators.py", line 30, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "django/contrib/admin/options.py", line 1718, in delete_view
    self.delete_model(request, obj)
  File "django/contrib/admin/options.py", line 1086, in delete_model
    obj.delete()
  File "django/db/models/base.py", line 896, in delete
    collector.delete()
  File "django/db/models/deletion.py", line 314, in delete
    sender=model, instance=obj, using=self.using
  File "django/dispatch/dispatcher.py", line 189, in send
    response = receiver(signal=self, sender=sender, **named)
  File "onadata/apps/logger/models/instance.py", line 97, in update_xform_submission_count_delete
    .get(pk=xform.user.profile.pk)
  File "django/db/models/fields/related.py", line 614, in __get__
    rel_obj = qs.get()
  File "django/db/models/query.py", line 334, in get
    self.model._meta.object_name
```

**Comment by jnm** [comment]
@noliveleger, does your fix for kobotoolbox/kpi#2509 fix this as well?

**Comment by noliveleger** [comment]
Indeed!
